src/sinks/aws_kinesis_streams/request_builder.rs

```diff
@@ -77,7 +87,7 @@ impl ByteSizeOf for KinesisRequest {
 }
 
 impl RequestBuilder<KinesisProcessedEvent> for KinesisRequestBuilder {
-    type Metadata = Metadata;
+    type Metadata = (KinesisMetadata, RequestMetadataBuilder);
```

Comment:
Indeed! Astute observation. When implementing this I seemed to have just been looking at the implementations of `RequestBuilder` instead of opening up the file with the trait definition to see if modifications there would make sense. Good lesson.
I'm about half way through implementing that improvement, will continue it tomorrow.